crates/polars-core/src/frame/group_by/into_groups.rs

```diff
@@ -185,6 +185,14 @@ where
                 };
                 num_groups_proxy(ca, multithreaded, sorted)
             },
+            #[cfg(feature = "dtype-decimal")]
+            DataType::Decimal(_, _) => {
+                // Safety: DecimalChunked's first field is ChunkedArray<Int128Type>
```

Comment:
We need this more often. Can you add a `to_primitive` on `DecimalChunked`. I want these pointer casts on a single location.

Reply:
`self` is `ChunkedArray<T>`, and we are dispatching based on `dtype()`. `DecimalChunked::to_primitive` doesn't work here, right? 
I've changed the comment as others' `// convince the compiler that we are this type.` for less confusing.

Reply:
Right, I see.